test/e2e/FlashLoanTracker.t.sol

```diff
@@ -181,4 +168,89 @@ contract E2eFlashLoanTracker is Test {
         require(Constants.AWETH.balanceOf(user) == 0);
         require(Constants.AWETH.balanceOf(address(Constants.SETTLEMENT_CONTRACT)) >= 10 ether);
     }
+
+    function test_10WethCollat100UsdsSwappingCollateralForWbtcWithFlashLoan() external {
+        AaveSetup.supplyWETH(vm, user, 100 ether);
+        uint256 _loanedAmount = 100 ether;
+        AaveSetup.prepareBorrow(vm, user, address(Constants.USDS), _loanedAmount);
+        uint256 _flashloanAmount = 1 * 10 ** 8;
+
+        // Setup the interactions
+        // 0) Send wbtc from borrower to tracker
+        // 1) Approve FlashLoanTracker's wbtc to AAVE's pool
+        // 2) Deposit wbtc into user's aWBTC
+        // 3) Mock a swap by pulling user funds
+        // 4) Mock the swap by sending wbtc to the FlashLoanTracker
+        // 5) Approve FlashLoanTracker's WBTC to borrower
+        // 6) Pull assets from FlashLoanTracker to the borrower
+        // 7) Borrower needs to approve the pool so the flashloan tokens + fees can be pulled out
+        ICowSettlement.Interaction[] memory _interactions = new ICowSettlement.Interaction[](8);
+
+        // 0) Send WBTC from borrower to tracker
+        _interactions[0] =
+            CowProtocolInteraction.transferFrom(Constants.WBTC, address(borrower), address(tracker), _flashloanAmount);
+
+        // 1) Approve FlashLoanTracker's WBTC to AAVE's pool
+        _interactions[1] = CowProtocolInteraction.borrowerApprove(
+            IBorrower(address(tracker)), Constants.WBTC, address(AaveSetup.POOL), type(uint256).max
+        );
+
+        // 2) Deposit wbtc into user's aWBTC
+        _interactions[2] = CowProtocolInteraction.supplyToAave(
+            IFlashLoanTracker(address(tracker)),
+            address(AaveSetup.POOL),
+            address(Constants.WBTC),
+            _flashloanAmount,
+            user
+        );
+
+        // 3) Mock a swap by pulling user funds
+        vm.prank(user);
+        Constants.AWETH.approve(address(Constants.SETTLEMENT_CONTRACT), type(uint256).max);
+        _interactions[3] = CowProtocolInteraction.transferFrom(
+            Constants.AWETH, address(user), address(Constants.SETTLEMENT_CONTRACT), 100 ether
+        );
+
+        uint256 _relativeFlashFee = AaveSetup.POOL.FLASHLOAN_PREMIUM_TOTAL();
+        uint256 _absoluteFlashFee = _loanedAmount * _relativeFlashFee / 1000;
+        uint256 _borrowedPlusFee = _loanedAmount + _absoluteFlashFee;
+
+        // 4) Mock the swap by sending wbtc to the FlashLoanTracker
+        deal(address(Constants.WBTC), address(Constants.SETTLEMENT_CONTRACT), _borrowedPlusFee);
+        _interactions[4] = CowProtocolInteraction.transfer(Constants.WBTC, address(tracker), _borrowedPlusFee);
```

Comment:
I'm not sure how hard it would be, but ideally we don't use an empty settle call but actually encode the user order as a trade (which can be fulfilled with swaps mimicking a private market maker solver). This way it's clear what the user order would.

From what I can see here, the user would place an order with the recipient being the FlashLoanTracker. I think this is unsafe, because - if for whatever reason the user has enough aWETH collateral (e.g. because ETH moons) - a solver could execute the user trade without depositing new WBTC collateral and thus drain the user funds into the FlashLoanTracker.